src/lightning-address.ts

```diff
@@ -71,14 +71,22 @@ export default class LightningAddress {
     return `https://${this.domain}/.well-known/keysend/${this.username}`;
   }
 
-  async generateInvoice(url: URL): Promise<Invoice> {
-    const data = await fetch(url);
-    const json = await data.json();
-    const paymentRequest = json && json.pr && json.pr.toString();
+  async generateInvoice(url: URL, params: { amount: string, comment?: string, nostr?: string }): Promise<Invoice> {
```

Comment:
we should have this generic and I also need to be able to create invoices that have payerdata etc. 

let the user just pass in anything